### PR TITLE
macOS: Fix undo redo not working on macOS.

### LINKED ISSUE #866: Undo not working in compose box on macOS
EDIT: Observed only on macOS so far. Seems not to affect Linux; untested on Windows.

I (on Zulip desktop 4.0.0) and another member of the Recurse Center noticed that undo isn't working in the compose box on macOS, either from the Edit menu or by using command-Z. Immediately after typing some text, if you try to undo, your input doesn't change at all.

I'm not sure if this is a bug, or a feature that hasn't been introduced yet.

### PR DESCRIPTION
**Pull request by akashnimare**
The default API provided by Electron doesn't work
as expected. More info here -
https://github.com/electron/electron/issues/15728

Fixes: #866.

---

**What's this PR do?**

**Any background context you want to provide?**

**Screenshots?**

**You have tested this PR on:**
  - [ ] Windows
  - [ ] Linux/Ubuntu
  - [x] macOS

### CODE REVIEW

**Comment by vsvipul** [comment]
LGTM! Can't test this though as I don't have a mac.

**Comment by andersk** [comment]
electron/electron#15728 was reported with Electron 3. Did you confirm that this is still an issue with Electron 8?

**Comment by akashnimare** [comment]
> [electron/electron#15728](https://github.com/electron/electron/issues/15728) was reported with Electron 3. Did you confirm that this is still an issue with Electron 8?

@andersk yeah, it's still broken. Let's merge?

File: app/renderer/js/main.ts [hunk 1]
```diff
@@ -739,6 +739,14 @@ class ServerManagerView {
 		return !(url.endsWith('/login/') || this.tabs[tabIndex].webview.loading);
 	}
 
+	getActiveWebview(): any {
```
Comment (andersk) [inline app/renderer/js/main.ts]: `any` → `Electron.WebviewTag`

File: app/renderer/js/main.ts [hunk 2]
```diff
@@ -739,6 +739,14 @@ class ServerManagerView {
...
+		const selector = 'webview:not([class*=disabled])';
+		const webview = document.querySelector(selector);
+		if (webview) {
```
Comment (andersk) [inline app/renderer/js/main.ts]: Why the `if`?

File: app/renderer/js/main.ts [hunk 3]
```diff
@@ -739,6 +739,14 @@ class ServerManagerView {
 		return !(url.endsWith('/login/') || this.tabs[tabIndex].webview.loading);
 	}
 
+	getActiveWebview(): any {
+		const selector = 'webview:not([class*=disabled])';
```
Comment (andersk) [inline app/renderer/js/main.ts]: `'webview:not(.disabled)`?

**Review by andersk — COMMENTED** [review]
I guess so. We should also get the Electron issue reopened or open a new one.

File: app/renderer/js/main.ts (hunk 2 above)
Comment (akashnimare) [reply app/renderer/js/main.ts]: This is just to be sure that we get the webview. Can also remove this.

File: app/renderer/js/main.ts (hunk 3 above)
Comment (akashnimare) [reply app/renderer/js/main.ts]: Yup, let me update it!

**Comment by akashnimare** [comment]
@andersk made some changes.

**Comment by adityamudgil2505** [comment]
I think the undo-redo issue is being solved. If there is any remaining work @akashnimare please share it with me.

**Comment by akashnimare** [comment]
@adityamudgil2505 can you get the Electron issue (https://github.com/electron/electron/issues/15728) reopened or maybe open a new one since it still doesn't work on v8?

**Pull request merged**